app/services/transaction_search/order_status_searcher.rb

```diff
@@ -3,10 +3,8 @@ module TransactionSearch
   class OrderStatusSearcher < BaseSearcher
 
     def options
-      # Unlike the other lookups, this query is much faster this way than using a subquery
-      OrderStatus.find_by_sql(order_details.joins(:order_status)
-                                                         .select("distinct(order_statuses.id), order_statuses.facility_id, order_statuses.name, order_statuses.lft")
-                                                         .reorder("order_statuses.lft").to_sql)
+      OrderStatus.where(id: order_details.select("distinct order_details.order_status_id"))
```

Comment:
It's a significant difference.

Searching on All Transactions with no date filters, it's the difference of 1/4 second and 5 seconds. The queries are definitely different. I think we do need to spend a little more time on it.

Run 1:
Old:
```
OrderStatus Load (217.5ms)  SELECT distinct(order_statuses.id), order_statuses.facility_id, order_statuses.name, order_statuses.lft FROM "ORDER_DETAILS" INNER JOIN "ORDERS" ON "ORDERS"."ID" = "ORDER_DETAILS"."ORDER_ID" INNER JOIN "ORDER_STATUSES" ON "ORDER_STATUSES"."ID" = "ORDER_DETAILS"."ORDER_STATUS_ID" WHERE "ORDERS"."STATE" = 'purchased' AND "ORDERS"."FACILITY_ID" = 21  ORDER BY order_statuses.lft
```
New:
```
  OrderStatus Load (4989.1ms)  SELECT "ORDER_STATUSES".* FROM "ORDER_STATUSES" WHERE "ORDER_STATUSES"."ID" IN (SELECT distinct order_details.order_status_id FROM "ORDER_DETAILS" INNER JOIN "ORDERS" ON "ORDERS"."ID" = "ORDER_DETAILS"."ORDER_ID" WHERE "ORDERS"."STATE" = 'purchased' AND "ORDERS"."FACILITY_ID" = :a1)  ORDER BY order_statuses.lft  [["facility_id", 21]]
```

Run 2:
Old:
```
OrderStatus Load (177.7ms)  SELECT distinct(order_statuses.id), order_statuses.facility_id, order_statuses.name, order_statuses.lft FROM "ORDER_DETAILS" INNER JOIN "ORDERS" ON "ORDERS"."ID" = "ORDER_DETAILS"."ORDER_ID" INNER JOIN "ORDER_STATUSES" ON "ORDER_STATUSES"."ID" = "ORDER_DETAILS"."ORDER_STATUS_ID" WHERE "ORDERS"."STATE" = 'purchased' AND "ORDERS"."FACILITY_ID" = 21  ORDER BY order_statuses.lft
```
New:
```
  OrderStatus Load (4204.8ms)  SELECT "ORDER_STATUSES".* FROM "ORDER_STATUSES" WHERE "ORDER_STATUSES"."ID" IN (SELECT distinct order_details.order_status_id FROM "ORDER_DETAILS" INNER JOIN "ORDERS" ON "ORDERS"."ID" = "ORDER_DETAILS"."ORDER_ID" WHERE "ORDERS"."STATE" = 'purchased' AND "ORDERS"."FACILITY_ID" = :a1)  ORDER BY order_statuses.lft  [["facility_id", 21]]
```